docs/docs/tutorials/orca-yarn-tutorial.md

```diff
@@ -0,0 +1,524 @@
+    ```python
+    init_orca_context(cluster_mode, cores, memory, num_nodes, 
+                      driver_cores, driver_memory, extra_python_lib, 
+                      conf={"spark.executorEnv.ARROW_LIBHDFS_DIR":"/opt/cloudera/parcels/CDH-${CHD_VERSION}/lib64/"})
+    ```
+
+### 5.1.2 Yarn Cluster
+Please call `init_orca_context` to create an OrcaContext at the very beginning of each Orca program.
+```python
+from bigdl.orca import init_orca_context
+
+if args.cluster_mode.startswith("yarn"):
+    if args.cluster_mode == "yarn-client":
+        ...
+    elif args.cluster_mode == "yarn-cluster":
+        init_orca_context(cluster_mode="yarn-cluster", cores=4, memory="10g", num_nodes=2,
+                          driver_cores=2, driver_memory="4g",
+                          extra_python_lib="model.py")
+```
+Run the example following command below:
+```bash
+python train.py --cluster_mode yarn-cluster --remote_dir hdfs://url:port/path/to/remote/data
+```
+* `--cluster_mode`: set the cluster_mode in `init_orca_context`.
+* `--remote_dir`: directory on a distributed storage for the dataset (see section 3) and saving the model.
+
+__Note__:
+* Please refer to __Section 4__ for the description of `extra_python_lib`.
+* For CDH users, please set `ARROW_LIBHDFS_DIR` (see __Section 2.3__) as a Spark conf in `init_orca_context`.
+    ```python
+    init_orca_context(cluster_mode, cores, memory, num_nodes, driver_cores, driver_memory, extra_python_lib, conf={"spark.executorEnv.ARROW_LIBHDFS_DIR":"/opt/cloudera/parcels/CDH-${CHD_VERSION}/lib64/"})
+    ```
+
+### 5.1.3 Jupyter Notebook
+You can simply run the example in a Jupyter Notebook. 
+
+```bash
+# Start a jupyter notebook
+jupyter notebook --notebook-dir=/path/to/notebook/directory --ip=* --no-browser
+```
+You need to migrate the example to notebook. Please call `init_orca_context` at the very beginning of the program and set the `cluster_mode` to `yarn-client`.
+```python
+from bigdl.orca import init_orca_context
+
+init_orca_context(cluster_mode="yarn-client", cores=4, memory="10g", num_nodes=2, 
+                  driver_cores=2, driver_memory="4g",
+                  extra_python_lib="model.py")
+```
+__Note:__
+* Jupyter Notebook cannot run on `yarn-cluster`, as the driver is not running on the __Client Node__(the notebook page).
+
+## 5.2 Use `bigdl-submit`
+BigDL provides an easy-to-use `bigdl-submit` script (under `${BIGDL_HOME}/scripts`), which could automatically setup configuration and jars files from the current activate Conda environment.
+
+On the __Client Node__ (where you submit applications), before submitting the example:
+1. Please call `init_orca_context` at the very beginning of the program.
+    ```python
+    from bigdl.orca import init_orca_context
+
+    if args.cluster_mode == "bigdl-submit":
+        init_orca_context(cluster_mode="bigdl-submit")
+    ```
+2. Pack the Conda environment to an archive.
+    ```bash
+    conda pack -o environment.tar.gz#environment 
```

Comment:
(hk-spark-submit) [kai@Almaren-Node-107 mastercard]$ conda pack -o environment.tar.gz#environment
Collecting packages...
CondaPackError: Unknown file extension 'environment.tar.gz#environment'

Shouldn't it just be `conda pack -o environment.tar.gz`?

Reply:
that's a comment but not one of the command, it will be removed.